core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -188,7 +188,7 @@ protected void cleanUncommitted(Set<ManifestFile> committed) {
         }
       }
 
-      this.newManifests = committedNewManifests;
```

Comment:
@jasonf20, I think that this should actually not modify the `newManifests` field at all and that this is a bug.

When a transaction retries, all of the the new manifests should be reused. In fact, in the initial version with just one new manifest, it was [left unchanged](https://github.com/apache/iceberg/pull/6335/files#diff-0ce77e454fbf0eed71c932cb352fb60f405d5ceb636950f3c2af36c6f93a4f48L181-L183). That way when the operation is re-committed the same set of manifests is reused. (The transaction is smart enough to only delete the files deleted by individual operations if the transaction commit succeeds.)

Setting this to `null` would fix the problem by causing Iceberg to rewrite the manifests every time the transaction is attempted. That works, but the extra writes are unnecessary.

Reply:
Good catch I just applied the same pattern from `newDataFilesAsManifests` and didn't notice this difference. 
Let me know if you want me to fix this or if you think this problem should be fixed by validation instead. i.e. should re-calling commit be allowed in user code?